### ISSUE TITLE
Support domain-local links and relative paths when crawling

### ISSUE THREAD

**Issue by lilith** [endymion/link-checker#6]
This seems like a bug to me - shouldn't internal link correctness be more important than external link correctness?

**Comment by mttrb** [comment]
I believe that the link-checker was originally developed to check links in Octopress blogs.  Octopress is a static blog site generator so it is unlikely that internal links would be broken as the pages and links are generated programmatically.  Obviously if you are using the link-checker in a different environment then your link checking priorities may be different.

**Comment by gravitystorm** [comment]
If this is intended behaviour, then there should be a big warning about it in the readme! I realise now that the introduction says "external links" a number of times but the subtlety escaped me before.

As for Octopress, any blog post could link to another internal blog post, so that's not really any different from a jekyll site or anything else. There's still a chance that your internal link is wrong (e.g. typo) and that it won't get picked up by this tool.

**Comment by endymion** [comment]
I agree, it needs to support relative links.  That has already been fixed, I just need to push out a new version.  I've been having problems dealing with issues and pull requests for this project because of a little baby who sucks up all of my time, but I'll try to merge in the remaining pull requests and release a new version this weekend.

I built this gem because I needed something for checking links in Octopress/Jekyll sites, but I always intended it to be able to scan any kind of site.  It's not intended to be tied specifically to Octopress.

**Comment by gravitystorm** [comment]
@endymion awesome - I look forward to the new version, and understand completely about other time commitments.
